packages/kbn-ai-playground/components/open_ai_key_flyout.tsx

```diff
@@ -0,0 +1,92 @@
+import { EuiButton, EuiButtonEmpty, EuiFieldText, EuiFlexGroup, EuiFlexItem, EuiFlyout, EuiFlyoutBody, EuiFlyoutFooter, EuiFlyoutHeader, EuiFormRow, EuiLink, EuiSpacer, EuiText, EuiTitle } from "@elastic/eui";
+import { i18n } from "@kbn/i18n";
+import React, { useState } from "react";
+
+export interface OpenAIKeyFlyOutProps {
+  onClose: () => void;
+  setOpenAIKey: (key: string) => void;
+}
+
+export const OpenAIKeyFlyOut: React.FC<OpenAIKeyFlyOutProps> = ({ onClose, setOpenAIKey }: OpenAIKeyFlyOutProps) => {
+  const [apiKey, setApiKey] = useState<string>('');
```

Comment:
I'm afraid we need to use useForm here, and check also isDirty value. 
And allow to get props of default value for apiKey, because when it's open twice it suppose to be there
I can help you with that